proposals/wg-image-compatibility.md

```diff
@@ -0,0 +1,106 @@
+
+
+The described incompatibility problem affects distributed systems.
+In multi-cloud, hybrid cloud, or even private cloud scenarios, one containerized cloud-native application will need to run
+on different Kubernetes distributions that use different host operating systems on different worker nodes.
+
+Operating systems can use different kernel versions with different configurations.
+Different operating system distributions may integrate different devices and their corresponding drivers,
+even for the same device different versions of drivers may be used. Therefore, different operating system distributions are not identical.
+Even within the same OS distro, different releases are also not identical.
+
+An image compatibility specification would help container image authors describe compatibility requirements in a standard way.
+The specification will be uploaded with the image to the image registry. This makes hard container compatibility requirements discoverable,
+programmable, and will support different consumers and cover use cases where the application requires a specific compatible environment.
+
+For further reference please follow:
+* [OCI Image Compatibility - the spec reasoning](https://docs.google.com/presentation/d/1F9GnCm2sULuyTJ5BEFZlL8Qjab81DK7g9Oy6qBfe5Qs/edit?usp=sharing)
+* [Revisit container and host OS compatibility](https://docs.google.com/document/d/1lzwh8DGMu5vXXHwJmnewYIMffkcOEvH8owX4UYjRcw0/edit#heading=h.fcxt9vheg92c)
+
+## Scope
+
+* Define image compatibility spec that describes special host OS requirements of containerized application.
+* The spec should describe container requirements for Linux, FreeBSD and Windows.
+* The final list of supported fields (like kernel modules, configuration, out-of-tree drivers etc.)
+  is to be determined by the working group for each supported OS.
+* Make the change as minimally invasive as possible to the image, runtime and distribution specs - that could be achieved over
+  delivering compatibility artifact type.
+* Deprecate os.version and os.features fields from the image spec that were mainly used for Windows.
+
+## Out of Scope
+
+* Change images build or container execution flow.
+* Introduce a new hard requirement to build images or run containers.
+
+## Intended work product
+
+* The result of this WG should be extended or a new OCI specification that allows containers to describe host OS requirements.
+* The specification will provide a much better experience for users to find out about container requirements against host OS, if those exist.
+* The specification will provide programmatic data for various OS operators that could influence configuration of nodes.
+
+## Proposed Owners
+
+The following have agreed to participate in the working group, review progress, report progress back to the OCI community, and present the results to the TOB once the working group has completed its objectives.
```

Comment:
Is there anything anywhere about frequency of meetings, times, etc.? That would probably determine my ability to participate in the short / long run.

Reply:
I think we should meet once a week. As for the time... it depends on the members of the working group, we will have to find the best hour. Once we have a complete list of owners, we will need to find a suitable time together.

Reply:
the description in the template also make me confused, there are many ways to contribute in the working group, meeting discussion, mail-list discussion, patches review, coding, documentation. Not all contributors want to have the responsibility to report progress back to the OCI community, and present the results to the TOB. If someone only want to participate in meeting discussion, can such contributors be listed in the proposed owers. Maybe the template can be updated that the working group elected leader to take the report responsibility? @sudo-bmitch

Reply:
I've been using this as a list of people willing to meet and work on the effort. It's mainly a commitment of time. There's always the risk that some will have a change of focus or meeting conflicts, and others will join in. Things I'm looking for in the list are a variety of representatives (not everyone from a single company) and representatives from the various specs that may be impacted.

I don't believe the TOB ever formally met for the referrers WG, we just gave periodic updates on the regular Thursday dev call, and then gave a final presentation there before submitting PRs. If a new spec is needed by the TOB, then the TOB can figure out what makes sense for that (I could see us using the Thursday call, posting the recording, and doing an async vote).